ClientCore/ClientConfiguration.cs

```diff
@@ -188,7 +192,10 @@ public void RefreshSettings()
 
         public string[] RecommendedResolutions => clientDefinitionsIni.GetStringValue(SETTINGS, "RecommendedResolutions", "1280x720,2560x1440,3840x2160").Split(',');
 
-        public string WindowTitle => clientDefinitionsIni.GetStringValue(SETTINGS, "WindowTitle", string.Empty);
+#pragma warning disable CNCNET0001 // L10N Failure
```

Comment:
> Amazing work! There's some things to fix, mostly stylistical nitpicks.
> 
> On the `CNCNET0001` warnings, they seem to be disabled in quite many places. If it needs to be disabled so often, is there any benefit to having that warning in the first place?

Most `L10N` calls are for literal strings. When compiling the code, source generator will check and enforce the following condition:
- The string is a literal string. Its value can be determined at compile-time. (to prevent improper `L10N` calls by someone who is not familiar to the localization method)
- The string is not trimmble.
- If the key is defined more than once, the values must be the same.

If we don't explicitly suppress `CNCNET0001`, we have to define another call that does the same thing as `L10N` with a different method name.

Reply:
So currently `CNCNET0001` has to be suppressed for all non-literal strings? That sounds rather impractical to me.

I think we should rather either disable the warning in such a case (which currently is a completely normal use case of L10N, as you can see from the large bunch of disabled `CNCNET0001` warnings) or introduce another method for non-literal strings.

Reply:
> So currently `CNCNET0001` has to be suppressed for all non-literal strings? That sounds rather impractical to me.
> 
> I think we should rather either disable the warning in such a case (which currently is a completely normal use case of L10N, as is evident from the large bunch of disabled `CNCNET0001` warnings) or introduce another method for non-literal strings.

The former is not acceptable as there would be no way to prevent less-experienced developers from writing something like `$“The output is {text}”.L10N("key-name")` or `"string.Format("whatever {0}", val).L10N("key-name")"`.

The latter is the only choice in order to dismiss the suppression comment. `L10N_Dynamic()` or something.

Reply:
I think code reviews could catch the clear majority of improper `L10N` use. But how you solve it is up to you and @Metadorius. As-is I don't think it's acceptable.

Reply:
@Metadorius https://github.com/Metadorius/xna-cncnet-client/pull/10

Reply:
@Rampastring what are your thoughts on just changing the warning level to a suggestion?

Reply:
I disagree. The point is to remind people. If the message is a `info` and is not suppressed, developers will ignore more messages as these messages are already shown before they make changes.

Reply:
That's a developer problem that they ignore it.

Reply:
> That's a developer problem if they ignore it.

How can it be the developer's problem when the code already has CNCNET0001 messages? Does it make sense to let the developer pay attention to how many time CNCNET0001 messages show, and only if the number is greater than the original count means they have a problem? Ridiculous to me.

Reply:
Well firstly your point wasn't very understandable in your previous message.

Secondly, I think you overestimate the importance of this aspect. If a person implements some new translated string, the message will notify them of the behavior. If they ignore it carelessly - that's their problem.

Reply:
More so, code analysis doesn't run on the whole solution by default, so it will mostly display only active document messages.

Reply:
> Secondly, I think you overestimate the importance of this aspect. If a person implements some new translated string, the message will notify them of the behavior. If they ignore it carelessly - that's their problem.

Good codes express itself. A message is by default ignorable if messages of the same type are prompted in the original codes and no one suppresses them. Leaving the CNCNET0001 messages means we have to write docs to explain in which case a message should be ignored or not ignored. Developers have also to read this doc before making any code changes. Is it worth than making a `L10N_Dynamic` method?

Reply:
Developers just have to read the *message* that appears, docs are not necessary in this case. Introducing another method that does an exact same thing just for the analyzer to shut up is an even more weird solution, and I am opposed to that solution.

Reply:
After considering this for a while, I've come to the conclusion that I support @Metadorius' view on this one. It seems unlikely that developers would mistakenly use non-literal strings with L10N when unnecessary, and when it happens, it'll most likely be caught on code reviews.

So I'd say either change `CNCNET0001` to info level or remove it completely.

Reply:
I am still against changing `CNCNET0001` to info. I made changes to the source generator so that it would try to recognize INI strings for the best and suppress warnings in such a situation. For other messages those are warnings. https://github.com/Metadorius/xna-cncnet-client/pull/10  @Metadorius

Reply:
@SadPencil first things first, I checked how you use that `CNCNET0001` diagnostic and ugh, you should absolutely not use one diagnostic for *four* different issues. Make a new diagnostic per each kind of an issue with an appropriate diagnostic level depending on the issue. Second, I'll review it later, but I really don't understand the fixation on this. A more helpful thing to do would be implementing localization of XNAMultiColumnListBox column widths and parameters.